src/main/java/jenkins/plugins/openstack/compute/JCloudsSlave.java

```diff
@@ -214,6 +214,33 @@ private static Openstack getOpenstack(String cloudName) {
         return JCloudsCloud.getByName(cloudName).getOpenstack();
     }
 
+    public JCloudsSlaveTemplate getTemplate() {
+        JCloudsCloud cloud = JCloudsCloud.getByName(cloudName);
+        Server server = cloud.getOpenstack().getServerById(getServerId());
+        String templateName = server.getMetadata().get(JCloudsSlaveTemplate.OPENSTACK_TEMPLATE_NAME_KEY);
+        return cloud.getTemplate(templateName);
+    }
+
+    /**
+     * Should this node be retained to meet the minimum instances constraint?
+     */
+    public boolean shouldBeRetained() {
```

Comment:
I do not think this methods belongs to node. Would it not be better to keep it in whatever abstraction that enforces the node numbers?